models/xception/xception_impl.hpp

```diff
@@ -0,0 +1,227 @@
+    includeTop = other.includeTop;
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+XceptionType<MatType>&
+XceptionType<MatType>::operator=(XceptionType&& other)
+{
+  if (this != &other)
+  {
+    ann::MultiLayer<MatType>::operator=(std::move(other));
+    numClasses = std::move(other.numClasses);
+    includeTop = std::move(other.includeTop);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+template<typename Archive>
+void XceptionType<MatType>::serialize(
+    Archive& ar, const uint32_t /* version */)
+{
+  ar(cereal::base_class<ann::MultiLayer<MatType>>(this));
+
+  ar(CEREAL_NVP(numClasses));
+  ar(CEREAL_NVP(includeTop));
+}
+
+template<typename MatType>
+void XceptionType<MatType>::SeparableConv(
+    ann::MultiLayer<MatType>* block,
+    const size_t inMaps,
+    const size_t outMaps,
+    const size_t kernelSize,
+    const size_t stride = 1,
+    const size_t padding = 0,
+    const bool useBias = false)
+{
+  block->template Add<ann::GroupedConvolution>(inMaps, kernelSize, kernelSize,
+    inMaps, stride, stride, padding, padding, "none", useBias);
+  block->template Add<ann::Convolution>(outMaps, 1, 1, 1, 1, 0, 0, "none",
+    useBias);
+}
+
+template<typename MatType>
+void XceptionType<MatType>::Block(
+    const size_t inMaps,
+    const size_t outMaps,
+    const size_t reps,
+    const size_t strides = 1,
+    const bool startWithRelu = true,
+    const bool growFirst = true)
+{
+  ann::MultiLayer<MatType>* block = new ann::MultiLayer<MatType>();
+  size_t filter = inMaps;
+  if (reps < 2)
+  {
+    if (startWithRelu)
+      block->template Add<ann::ReLU>();
+    SeparableConv(block, inMaps, outMaps, 3, 1, 1, false);
+    block->template Add<ann::BatchNorm>();
+  }
+  else
+  {
+    if (growFirst)
+    {
+      if (startWithRelu)
+        block->template Add<ann::ReLU>();
+      SeparableConv(block, inMaps, outMaps, 3, 1, 1, false);
+      block->template Add<ann::BatchNorm>();
+      filter = outMaps;
+    }
+    if (startWithRelu || growFirst)
+      block->template Add<ann::ReLU>();
+    SeparableConv(block, filter, filter, 3, 1, 1, false);
+    block->template Add<ann::BatchNorm>();
+    if (reps > 2)
+    {
+      for (size_t i = 0; i < reps - 2; i++)
+      {
+        block->template Add<ann::ReLU>();
+        SeparableConv(block, filter, filter, 3, 1, 1, false);
+        block->template Add<ann::BatchNorm>();
+      }
+    }
+    if (!growFirst)
+    {
+      block->template Add<ann::ReLU>();
+      SeparableConv(block, inMaps, outMaps, 3, 1, 1, false);
+      block->template Add<ann::BatchNorm>();
+    }
+  }
+  if (strides != 1)
+  {
+    block->template Add<ann::Padding>(1, 1, 1, 1);
+    block->template Add<ann::MaxPooling>(3, 3, strides, strides);
+  }
+  if (inMaps != outMaps || strides != 1)
+  {
+    ann::MultiLayer<MatType>* block2 = new ann::MultiLayer<MatType>();
+    block2->template Add<ann::Convolution>(outMaps, 1, 1, strides, strides,
+      0, 0, "none", false);
```

Comment:
```suggestion
        0, 0, "none", false);
```
I might have missed some of these, but here's another one I saw.